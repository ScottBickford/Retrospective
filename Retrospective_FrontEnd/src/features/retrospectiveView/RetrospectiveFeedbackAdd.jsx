import React, { Fragment, useState } from 'react';

const RetrospectiveFeedbackAdd = ({ handleAddFeedback }) => {
  const [name, setName] = useState('');
  const [body, setBody] = useState('');
  const [feedbackType, setFeedbackType] = useState('');
  const [error, setError] = useState(null);

  const handleSubmit = event => {
    event.preventDefault();
    setError(null);
    handleAddFeedback({ name, body, feedbackType })
      .then(response => {
        clearFields();
      })
      .catch(error => {
        setError(error);
      });
  };

  const canSubmit = () => {
    return name.trim() !== '' && body.trim() !== '' && feedbackType !== '';
  };

  const clearFields = () => {
    setName('');
    setBody('');
    setFeedbackType('');
  }

  const feedbackTypeOptions = () => {
    const options = ['', 'Positive', 'Negative', 'Idea', 'Praise'];
    return options.map(o => (
      <option key={o} value={o}>
        {o}
      </option>
    ));
  };

  return (
    <Fragment>
      <h4>Add feeback</h4>
      {error && <div style={{ color: 'red' }}>{error.message}</div>}
      <form onSubmit={handleSubmit}>
        <div>
          <div>Name of person:</div>
          <input
            type='text'
            style={{ width: 200 }}
            maxLength='200'
            value={name}
            onChange={e => setName(e.target.value)}
          />
        </div>
        <div>
          <div>Body:</div>
          <input
            type='text'
            style={{ width: 200 }}
            maxLength='1000'
            value={body}
            onChange={e => setBody(e.target.value)}
          />
        </div>
        <div>
          <div>Feedback type:</div>
          <select
            type='text'
            style={{ width: 200 }}
            maxLength='1000'
            value={feedbackType}
            onChange={e => setFeedbackType(e.target.value)}
          >
            {feedbackTypeOptions()}
          </select>
        </div>
        <button style={{marginTop:10}} disabled={!canSubmit()}>Submit</button>
      </form>
    </Fragment>
  );
};

export default RetrospectiveFeedbackAdd;
