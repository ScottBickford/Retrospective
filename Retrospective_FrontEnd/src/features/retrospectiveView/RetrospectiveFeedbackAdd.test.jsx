import React from 'react';
import renderer from 'react-test-renderer';
import RetrospectiveFeedbackAdd from './RetrospectiveFeedbackAdd';

describe('RetrospectiveFeedbackAdd', () => {
  test('snapshot renders', () => {    
    const component = renderer.create(<RetrospectiveFeedbackAdd />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});