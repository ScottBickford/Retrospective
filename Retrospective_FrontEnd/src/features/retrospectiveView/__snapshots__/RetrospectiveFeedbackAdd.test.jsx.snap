// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrospectiveFeedbackAdd snapshot renders 1`] = `
Array [
  <h4>
    Add feeback
  </h4>,
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        Name of person:
      </div>
      <input
        maxLength="200"
        onChange={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
        type="text"
        value=""
      />
    </div>
    <div>
      <div>
        Body:
      </div>
      <input
        maxLength="1000"
        onChange={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
        type="text"
        value=""
      />
    </div>
    <div>
      <div>
        Feedback type:
      </div>
      <select
        maxLength="1000"
        onChange={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
        type="text"
        value=""
      >
        <option
          value=""
        >
          
        </option>
        <option
          value="Positive"
        >
          Positive
        </option>
        <option
          value="Negative"
        >
          Negative
        </option>
        <option
          value="Idea"
        >
          Idea
        </option>
        <option
          value="Praise"
        >
          Praise
        </option>
      </select>
    </div>
    <button
      disabled={true}
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Submit
    </button>
  </form>,
]
`;
